src/mono/mono/metadata/reflection.c

```diff
@@ -184,36 +184,20 @@ mono_reflected_hash (gconstpointer a) {
 static void
 clear_cached_object (MonoDomain *domain, gpointer o, MonoClass *klass)
 {
-	mono_domain_lock (domain);
-	if (domain->refobject_hash) {
-        ReflectedEntry pe;
-		gpointer orig_pe, orig_value;
+	MonoMemoryManager *memory_manager = mono_domain_ambient_memory_manager (domain);
 
-		pe.item = o;
-		pe.refclass = klass;
+	mono_mem_manager_lock (memory_manager);
 
-		if (mono_conc_g_hash_table_lookup_extended (domain->refobject_hash, &pe, &orig_pe, &orig_value)) {
-			mono_conc_g_hash_table_remove (domain->refobject_hash, &pe);
-			free_reflected_entry ((ReflectedEntry*)orig_pe);
-		}
-	}
-	mono_domain_unlock (domain);
-}
+	gpointer orig_pe, orig_value;
+	ReflectedEntry pe = { .item = o, .refclass = klass };
```

Comment:
I don't think the C++ build will like this (designated initializers are new in C++20, so with a new enough compiler, maybe...)

Reply:
Yeah, maybe. They're nice though...
Having to write C++-compatible C is crippling and I've become increasingly opinionated on this (on the side of "give up on c++ compat) after writing some pure C programs in my own time recently.

Reply:
The msvc build fails because of this.